ngraph/frontend/onnx_import/src/op/dropout.cpp

```diff
@@ -0,0 +1,137 @@
+//*****************************************************************************
+// Copyright 2017-2021 Intel Corporation
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//*****************************************************************************
+
+#include <memory>
+
+#include "core/null_node.hpp"
+#include "default_opset.hpp"
+#include "exceptions.hpp"
+#include "ngraph/log.hpp"
+#include "ngraph/node.hpp"
+#include "op/dropout.hpp"
+
+namespace ngraph
+{
+    namespace onnx_import
+    {
+        namespace op
+        {
+            namespace
+            {
+                OutputVector
+                    build_dropout(const Node& node, float drop_probability, bool training_mode)
+                {
+                    CHECK_VALID_NODE(
+                        node,
+                        drop_probability == 0 || !training_mode,
+                        "Training mode is not supported for Dropout op if drop_probability is not "
+                        "equal 0");
+
+                    const auto input_data = node.get_ng_inputs().at(0);
+                    const bool return_mask = node.get_outputs_size() > 1;
+
+                    if (return_mask)
+                    {
+                        NGRAPH_WARN << "Default mask for Dropout is ignored, "
+                                    << "because of unsupported constant networks";
+                        /*const auto mask = std::make_shared<default_opset::Broadcast>(
+                                default_opset::Constant::create(ngraph::element::boolean,
+                            Shape{}, {true}),
+                                std::make_shared<default_opset::ShapeOf>(input_data));*/
+                        // If constant network is supported mask should be returned instead of
+                        // NullNode (ticket 48055)
```

Comment:
I don't think that throwing a warn here is a good idea.
It can be used in the scenario when network isn't constant.

Reply:
@tsocha In this case (`drop_probability == 0 || !training_mode`) mask output will be always produced by const network. Warning will be removed after const network support introduction.

Reply:
Even when model which contains this Dropout will have 300 nodes?
I'm not sure about it.

Reply:
right, removed